library/core/src/sync/atomic.rs

```diff
@@ -3401,12 +3401,8 @@ macro_rules! atomic_int {
             /// Doing non-atomic reads and writes on the resulting integer can be a data race.
             /// This method is mostly useful for FFI, where the function signature may use
             #[doc = concat!("`*mut ", stringify!($int_type), "` instead of `&", stringify!($atomic_type), "`.")]
-            ///
-            /// Returning an `*mut` pointer from a shared reference to this atomic is safe because the
-            /// atomic types work with interior mutability. All modifications of an atomic change the value
-            /// through a shared reference, and can do so safely as long as they use atomic operations. Any
-            /// use of the returned raw pointer requires an `unsafe` block and still has to uphold the same
-            /// restriction: operations on it must be atomic.
+            /// All modifications of an atomic change the value through a shared reference, and can do so safely
+            /// as long as they use atomic operations.
```

Comment:
I don't get the first part of this sentence... what else would they do?

Reply:
Github is being silly and doesn't let me make this a suggestion, so let me post this manually -- how about:
```rust
            /// Note that doing non-atomic reads or writes on the resulting integer can be
            /// Undefined Behavior due to a data race; see the [memory model section] for further information.
            ///
            /// This method is mostly useful for FFI, where the function signature may use
            #[doc = concat!("`*mut ", stringify!($int_type), "` instead of `&", stringify!($atomic_type), "`.")]
```